src/QMCWaveFunctions/Fermion/MultiSlaterDeterminantFast.cpp

```diff
@@ -372,6 +371,42 @@ WaveFunctionComponent::PsiValueType MultiSlaterDeterminantFast::ratio(ParticleSe
   return curRatio;
 }
 
+void MultiSlaterDeterminantFast::evaluateRatios(const VirtualParticleSet& VP, std::vector<ValueType>& ratios)
```

Comment:
So I guess there is no unit test?
New code should have unit tests.

Reply:
The code is actually covered by deterministic tests diamondC_1x1x1-Gaussian_pp_MSD and diamondC_2x1x1-Gaussian_pp_MSD as I turned them on below for offload builds but codecov doesn't recognize it.

I agree that unit test will be helpful. I will expand test_multi_slater_determinant.

Reply:
Online a a few unit tests.

Reply:
I wouldn't say it's "good" that codecov doesn't recognize the deterministic system tests as "coverage" but application level tests of features are not the same as unit test code. They do almost nothing to describe expected behavior of particular units of code. Thanks for adding these.